app/libs/aws_ipi_clusters.py

```diff
@@ -141,6 +141,13 @@ def create_or_destroy_aws_ipi_cluster(
             s3_bucket_path=s3_bucket_path,
         )
 
+    if not res and not cleanup:
+        click.echo(
+            f"Failed to run cluster {action}\n\tERR: {err}\n\tOUT: {out}.\n\tCleaning leftovers."
+        )
+        create_or_destroy_aws_ipi_cluster(
+            cluster_data=cluster_data, action="destroy", cleanup=True
```

Comment:
What happens if the original action was already destroy? will it try to destroy again? since it might fail again, maybe call cleanup as a separate function?

Reply:
no need for separate function, we need to check if the `action` == `create`

Reply:
done